ddtrace/opentelemetry/tracer.go

```diff
@@ -17,15 +19,44 @@ import (
 var _ oteltrace.Tracer = (*oteltracer)(nil)
 
 type oteltracer struct {
-	cfg      oteltrace.TracerConfig
 	provider *TracerProvider
 	ddtrace.Tracer
 }
+type otelCtxToDDCtx struct {
+	oc oteltrace.SpanContext
+}
+
+func (c *otelCtxToDDCtx) TraceID() uint64 {
+	allTraceID := [16]byte(c.oc.TraceID())
```

Comment:
I think the cast is unnecessary. You should just be able to do this here and throughout:
```
id := c.oc.TraceID()
return binary.BigEndian.Uint64(id[8:])
```
(I'd also just suggest calling them `id` since the longer name doesn't add much extra information)